runtime/compiler/aarch64/runtime/PicBuilder.spp

```diff
@@ -246,7 +259,68 @@ _virtualUnresolvedHelper:
 	hlt	#0	// Not implemented yet
 
 _interfaceCallHelper:
-	hlt	#0	// Not implemented yet
+	stp	x0, x1, [J9SP, #-64]!
+	stp	x2, x3, [J9SP, #16]
+	stp	x4, x5, [J9SP, #32]
+	stp	x6, x7, [J9SP, #48]
+	mov	x3, x30					// preserve LR
+	add	x0, x30, #J9TR_UICSnippet_CP			// get CP/index pair pointer
+	ldr	x1, [x30, #J9TR_ICSnippet_returnAddress]	// get code cache RA
+	mov	x4, x1						// protect RA in x4 (in L_commonLookupException, it is expected)
+	bl	jitResolveInterfaceMethod			// call the helper
+	cbz	x0, L_commonLookupException			// if resolve failed, throw the exception
+	add	x0, x3, #J9TR_Snippet_CallInstruction		// get address of BL instruction in snippet
+	ldr	x1, const_interfaceCompeteSlot2			// get new snippet branch target
+	mov	x2, #TR_ARM64interfaceCompeteSlot2
+	bl	L_refreshHelper				// rewrite the BL
+	mov	x30, x3					// restore LR
+	ldr	x0, [J9SP, #0]					// refetch 'this'
+	b	L_continueInterfaceSend			// lookup interface method and send
+_interfaceCompeteSlot2:
+_interfaceSlotsUnavailable:
+	stp	x0, x1, [J9SP, #-64]!
+	stp	x2, x3, [J9SP, #16]
+	stp	x4, x5, [J9SP, #32]
+	stp	x6, x7, [J9SP, #48]
+L_continueInterfaceSend:
+#ifdef J9VM_GC_COMPRESSED_POINTERS
+	ldr	w0, [x0, #J9TR_ObjectHeader_class]		// load class offset of receiver
+#else
+	ldr	x0, [x0, #J9TR_ObjectHeader_class]		// load class of receiver
+#endif
+	and	x0, x0, #~(J9TR_RequiredClassAlignment-1)	// mask VFT bits
+	add	x1, x30, #J9TR_ICSnippet_InterfaceClass		// get InterfaceClass/MethodIndex pair pointer
+	ldr	x2, [x30, #J9TR_ICSnippet_returnAddress]	// get code cache RA
+	mov	x4, x2						// protect LR in x4 (in L_commonLookupException, it is expected)
+	bl	jitLookupInterfaceMethod			// call the helper
+	cbz	x0, L_commonLookupException			// if resolve failed, throw the exception
+	mov	x15, #J9TR_InterpVTableOffset
+	sub	x15, x15, x0					// convert interp vTableIndex to jit index (must be in x15 for patch virtual)
+	mov	x30, x4					// set LR = code cache RA
+	ldr	x0, [J9SP, #0]					// refetch 'this'
+#ifdef J9VM_GC_COMPRESSED_POINTERS
+	ldr	w14, [x0, #J9TR_ObjectHeader_class]		// load class offset of receiver
+#else
+	ldr	x14, [x0, #J9TR_ObjectHeader_class]		// load class of receiver
+#endif
+	and	x14, x14, #~(J9TR_RequiredClassAlignment-1)	// mask VFT bits
+	ldr	x1, [J9SP, #8]					// restore other parameter regs
+	ldp	x2, x3, [J9SP, #16]
+	ldp	x4, x5, [J9SP, #32]
+	ldp	x6, x7, [J9SP, #48]
+	add	J9SP, J9SP, #64
+	ldr	x15, [x14, x15]				// jump thru vtable
+	br	x15
+
+L_commonLookupException:
+	add	J9SP, J9SP, #64				// clean up stack but do not restore register values
+	ldr	x0, [J9VMTHREAD, #J9TR_VMThreadCurrentException]	// load pending exception from vmStruct
+	mov	x30, x15					// move correct LR in to get exception throw.
```

Comment:
From the three places in this code that can reach this exception throw (e.g., L271, L296), is x15 set up to something?  If so, how?

Reply:
It looks like I made a mistake in renaming registers.  Changing to use `x28` instead of `x4` and `x15`.